lib/matplotlib/gridspec.py

```diff
@@ -232,38 +232,42 @@ def _normalize(key, size, axis):  # Includes last index.
 
 class GridSpec(GridSpecBase):
     """
-    Specifies the geometry of the grid that a subplot can be placed in.
+    A grid layout to place subplots within a figure.
 
-    The location of grid is determined by similar way as the SubplotParams.
+    The location of the grid cells is determined in a similar way as
+    `~.figure.SubplotParams` using *left*, *right*, *top*, *bottom*, *wspace*
+    and *hspace*.
     """
     def __init__(self, nrows, ncols, figure=None,
                  left=None, bottom=None, right=None, top=None,
                  wspace=None, hspace=None,
                  width_ratios=None, height_ratios=None):
         """
-        The number of rows and number of columns of the grid need to be set.
-        Optionally, the subplot layout parameters (e.g., left, right, etc.)
-        can be tuned.
-
         Parameters
         ----------
         nrows, ncols : int
             The number of rows and columns of the grid.
 
         figure : `~.figure.Figure`, optional
+            Only used for constrained layout to create a proper layoutbox.
 
         left, right, top, bottom : float, optional
             Extent of the subplots as a fraction of figure width or height.
             Left cannot be larger than right, and bottom cannot be larger than
-            top.
+            top. If not given, the values will be inferred from a figure or
+            rcParams when necessary. See also `GridSpec.get_subplot_params`.
```

Comment:
i think the when necessary is extraneous

Reply:
What I wanted to express with that is that it is a delayed evaluation. In contrast to many other default fallbacks `GridSpec.left` stays *None*. The paradigm seems to be to use `get_subplot_params` which will resolve to the given alternatives.

Reply:
at run time or at draw time or when the figure is being rendered? Those are both a bit more jargony than I like but maybe closer to clearer?

Reply:
Thanks. Chosen "at draw time" in #15106.